diff --git a/Headers/AppKit/NSKeyValueBinding.h b/Headers/AppKit/NSKeyValueBinding.h
index 445867e..7a6ab89 100644
--- a/Headers/AppKit/NSKeyValueBinding.h
+++ b/Headers/AppKit/NSKeyValueBinding.h
@@ -109,6 +109,7 @@ APPKIT_EXPORT NSString *NSContentArrayBinding;
 APPKIT_EXPORT NSString *NSContentBinding;
 APPKIT_EXPORT NSString *NSContentDictionaryBinding;
 APPKIT_EXPORT NSString *NSContentObjectBinding;
+APPKIT_EXPORT NSString *NSContentSetBinding;
 APPKIT_EXPORT NSString *NSContentValuesBinding;
 APPKIT_EXPORT NSString *NSEditableBinding;
 APPKIT_EXPORT NSString *NSEnabledBinding;
diff --git a/Source/NSArrayController.m b/Source/NSArrayController.m
index 60ebab0..f37a6a6 100644
--- a/Source/NSArrayController.m
+++ b/Source/NSArrayController.m
@@ -147,6 +147,7 @@
   if (self == [NSArrayController class])
     {
       [self exposeBinding: NSContentArrayBinding];
+      [self exposeBinding: NSContentSetBinding];
       [self exposeBinding: NSSelectionIndexesBinding];
       [self setKeys: [NSArray arrayWithObjects: NSContentBinding, NSContentObjectBinding, nil]
 	    triggerChangeNotificationsForDependentKey: @"arrangedObjects"];
@@ -545,7 +546,8 @@ atArrangedObjectIndexes: (NSIndexSet*)idx
   withKeyPath: (NSString *)keyPath
       options: (NSDictionary *)options
 {
-  if ([binding isEqual: NSContentArrayBinding])
+  if ([binding isEqual: NSContentArrayBinding]
+    || [binding isEqual: NSContentSetBinding])
     {
       GSKeyValueBinding *kvb;
 
@@ -574,6 +576,10 @@ atArrangedObjectIndexes: (NSIndexSet*)idx
     {
       return [NSArray class];
     }
+  else if ([binding isEqual: NSContentSetBinding])
+    {
+      return [NSSet class];
+    }
   else
     {
       return [super valueClassForBinding: binding];
diff --git a/Source/externs.m b/Source/externs.m
index 25b7a3b..dd85deb 100644
--- a/Source/externs.m
+++ b/Source/externs.m
@@ -695,6 +695,7 @@ APPKIT_DECLARE NSString *NSContentArrayBinding = @"contentArray";
 APPKIT_DECLARE NSString *NSContentBinding = @"content";
 APPKIT_DECLARE NSString *NSContentDictionaryBinding = @"contentDictionary";
 APPKIT_DECLARE NSString *NSContentObjectBinding = @"contentObject";
+APPKIT_DECLARE NSString *NSContentSetBinding = @"contentSet";
 APPKIT_DECLARE NSString *NSContentValuesBinding = @"contentValues";
 APPKIT_DECLARE NSString *NSEditableBinding = @"editable";
 APPKIT_DECLARE NSString *NSEnabledBinding = @"enabled";
